udata/core/reuse/api.py

```diff
@@ -159,7 +160,7 @@ def post(self, reuse):
             api.abort(400, 'Expect a dataset identifier')
         try:
             dataset = Dataset.objects.get_or_404(id=request.json['id'])
-        except Dataset.DoesNotExist:
+        except (Dataset.DoesNotExist, ValidationError):
```

Comment:
I'm entirely convinced by catching `ValidationError` here or there. We may be missing many locations since `get_or_404` is used all around.
Shouldn't we modify the `to_python` to not raise a ValidationError if the value is not an ObjectId?